### ISSUE TITLE
MS SQL Connector

### ISSUE THREAD

**Issue by marktani** [prisma/prisma1#1642]
This feature requests serves as a central place to discuss development and progress for the MS SQL connector.

**Comment by SebaBoler** [comment]
When you can expect a working connector to Ms SQL ??

**Comment by iamgmd** [comment]
I need this too

**Comment by marktani** [comment]
Hey @SebaBoler and @iamgmd, thanks a lot for your interest in the MS SQL connector 🙂 
I would love to hear your thoughts on specific use cases you'd like to be covered, and an initial suggestion for the SDL syntax and exposed capabilities.

You can read more [contributing to a DB connector](https://github.com/graphcool/prisma/issues/1751) here.

**Comment by jakenuts** [comment]
Super excited about this, though I will probably fall back on the hosted platform as I'm getting tired of managing everything - the NET community is ~desperate for a GraphQL API interface to SQL Server.

**Comment by Sawtaytoes** [comment]
I noticed MS SQL was missing on the main README: https://github.com/graphcool/prisma.

**Comment by nadilas** [comment]
+1

**Comment by jurby** [comment]
+1

**Comment by boverdrive** [comment]
I waiting for this

**Comment by eabassey** [comment]
I need this badly

**Comment by marktani** [comment]
Thanks a lot for your feedback everyone 🙂

If you have any thoughts in the spirit of what I mentioned above, I'd be happy to hear it! 
If you want to express your desire in this feature, please use Github reactions on the original post. Thanks! 🙌

**Comment by edenprairie** [comment]
I really need this! Please make this happen!

**Comment by felixti** [comment]
+1

**Comment by albangiroux** [comment]
+1

**Comment by develomark** [comment]
There is a huge .NET community - by supporting MSSQL and writing some how-to's on deploying onto Azure I think you will capture some massive traction in the enterprise space - especially financial services.

**Comment by BrianLondon** [comment]
Big +1 on this. Regarding features even just reproducing the pgsql functionality would solve 90% of users needs.

**Comment by playerx** [comment]
This is important because there are lots of enterprise organizations as @develomark mentioned, which use MS SQL and they will be able to start using Prisma (I'll be one who will suggest them to try), hope it will happen soon :) Also I'll take a look current connectors and I'll try to share something valuable about that.

One note here, if you will support azure sql, ms sql will be supported also, because azure sql is limited version of ms sql, but those limits will not have any effects for Prisma.

**Comment by vuchl** [comment]
I also would like to take a vote for a MSSQL connector to
1. push Prisma/GraphQL to enterprise environments
2. broaden the availabitly and compatibility of Prisma

Since I have become a user of MSSQL I must say it's a great DBMS and I would like to start a GraphQL project with some of our customers. Therefore I would like to have a solid development library that fits with SQL Server and is capable of using some more of the TSQL features and is not limited to a a really small subset of the SQL standard.

One could say to go with .NET/.NET Core but making Prisma available as a JavaScript solution to the market would propably also accell the need and use of full stack JS developers that could create products for big enterprise customers.

**Comment by sytpb** [comment]
hope this .

**Comment by Gary-Shurgin** [comment]
When I execute 'prisma deploy' to a local MySQL instance, I get this error.
What am I missing?

TypeError: Cannot read property 'name' of undefined
    at Socket.<anonymous> (C:\Users\Owner\AppData\Roaming\npm\node_modules\prisma\node_modules\pg\lib\connection.js:123:21)

**Comment by iamgmd** [comment]
@Gary-Shurgin Not sure if your post is relevant in this thread as this one is related to wanting MS SQL in Prisma, nothing to do with MySQL.

**Comment by Patrick-Spiegel** [comment]
This connector should be shipped by default as soon as possible. 
Supported should be
- Tables (of course)
- Views
- Stored Procedures (including result selection)
- Table- and Scalarfunctions
- Table Types

**Comment by mycaule** [comment]
Can you support through JDBC https://mvnrepository.com/artifact/com.microsoft.sqlserver ?

Also need to consider supporting different versions
https://docs.microsoft.com/fr-fr/sql/connect/jdbc/microsoft-jdbc-driver-for-sql-server-support-matrix

**Comment by Ibrahim-Islam** [comment]
Would like to know whether the work on this has started or not? Dont see a branch for it either.

**Comment by PJCAfonso** [comment]
I need this so badly

**Comment by Verikon** [comment]
I notice there's some development on the JDBC connector. Having not worked on MSSQL for a long time, would I have to ask, does the JDBC connector make this effective for using MSQL? Azure SQL?

**Comment by doitlikepruett** [comment]
Anyone have an update on this? I would like to help contribute to the project if it's in active development.

**Comment by dnikl** [comment]
CRUD functions with API access would work at first. We would love to use Prisma. @marktani is there already something going on?

**Comment by dnikl** [comment]
> You can read more [contributing to a DB connector](https://github.com/graphcool/prisma/issues/1751) here.

404

**Comment by exonianp** [comment]
It will be great to have it.

**Comment by curbol** [comment]
I need support for Azure SQL

**Comment by tanekim77** [comment]
I need it too

**Comment by adilasif** [comment]
Is this being worked on? Don't see a branch that seems relevant. If it is, I'd love to get my hands dirty and help get it finished.

**Comment by MichaelKoch** [comment]
Hey .. 
pls. implement this connector to push prisma to enterprise environment

**Comment by codiemullins** [comment]
Is there a way to contribute to new connectors?  A comment mentions a doc on this topic but it’s a 404 and I don’t see any updated link.

**Comment by toindev** [comment]
https://github.com/prisma/prisma/issues/1751 for the new link

> 
> 
> > You can read more [contributing to a DB connector](https://github.com/graphcool/prisma/issues/1751) here.
> 
> 404

**Comment by tanekim77** [comment]
I need this so badly. Any news? Is this even started?

**Comment by JohnMilazzo** [comment]
Has there been any progress on this?  Is there any way to contribute?  A MSSQL connector is in much demand!   🤞

**Comment by muloka** [comment]
> Has there been any progress on this? Is there any way to contribute? A MSSQL connector is in much demand!

Its possible to build a GraphQL solution with Apollo and MSSQL. Its still not as elegant as using Prisma and would love to see a connector built for it. 

I used this article as a starting point: https://medium.com/infocentric/setup-a-graphql-api-with-apollo-2-0-sequelize-and-express-js-608d1365d776

Installed tedious instead of mysql2. Hope this helps.

**Comment by ilovejs** [comment]
you needa another round of funding then build it 
https://github.com/microsoft/mssql-jdbc

**Comment by superdom4real** [comment]
definitely waiting for it.

**Comment by tvvignesh** [comment]
Would be great if we have some idea on this - The organisation where I work at is investing on creating Microservices, APIs/GraphQL, etc. for its legacy infrastructure based on MS SQL. Wanted to know if we can leverage this at some point of time.

Btw, Amazing work by all of you guys with Prisma! Its even cooler than GraphCool

**Comment by iyeremuk** [comment]
Need this badly.

**Comment by AndryHTC** [comment]
Is it possible to ask if we are even near to a beta stage? We've followed this confidently from 1Y ago... But still nothing. I know even 2 more little companies are waiting for this for updating some old projects.

**Comment by infin8loop** [comment]
Just adding to the growing list of people wanting this feature and also wondering about the **status of current efforts** and the **potential to contribute** (link above is still 404).

**Comment by cbrunnkvist** [comment]
@infin8loop yes the team needs to provide better guides for _"how to contribute"_ and _"writing a db connector"_ in order to enable the community to help here

**Comment by angad777** [comment]
Do we have any timelines for this? Keen to see or contribute to this as it will greatly help when building for enterprises that work only with MS SQL due to legacy implementations.

**Comment by AlexSun98** [comment]
I need it badly too. any update yet?

**Comment by AndryHTC** [comment]
I don't understand why no one reply. I understand that actually there aren't any intention of doing some efforts; but I don't understand why no one is explicitly saying this.

**Comment by AlexSun98** [comment]
Typeorm or sequenlize could be alternatives solution. But they are not designed for gql specifically.

**Comment by AndryHTC** [comment]
This is annoying. MS Sql is still in the top 3 most used DBs world wide, double more used than Postgres that already has the connector.

Also if it not deserves a connector yet, I think may deserves a reply at least

**Comment by JesseHenson** [comment]
Is this due to the different type of connection. I see on W3 that you simply import mysql and connect, where as a msSql database needs an odbc connection. It would make sense then that a connection like this might take longer and work differently with the infrustructure. 

It would be a cool feature to have. Is there a way to use a node server as a middleware for this. Could we use a node server connecting through odbc then pass that to prisma directly from the query response? 

https://www.tutorialsteacher.com/nodejs/access-sql-server-in-nodejs

**Comment by AndryHTC** [comment]
@JesseHenson
All the kind of reasons would be accepted. A "This feature is postponed due to technical issues" would be great also...

The silence NO.

**Comment by AlexSun98** [comment]
if you really need sth for mssql database connector badly. There is another open source library called sql datasource, it’s been recommended by Apollo offical document(you should be able to find it y reading doc of Apollo-datasource-rest)You can find it from Github(https://github.com/cvburgess/SQLDataSource/blob/master/README.md#sqldatasource). It’s based on knex and Apollo-datasource. It provides huge flexibility to build a ms sql server based datasource connector at gql server, it has some good features such as batching, transactions and simple syntax. Try it with dataloader would be a way to go if you can’t wait prisma’s new release. Hope it helps.

**Comment by cpattonDev** [comment]
> > Has there been any progress on this? Is there any way to contribute? A MSSQL connector is in much demand!
> 
> Its possible to build a GraphQL solution with Apollo and MSSQL. Its still not as elegant as using Prisma and would love to see a connector built for it.
> 
> I used this article as a starting point: https://medium.com/infocentric/setup-a-graphql-api-with-apollo-2-0-sequelize-and-express-js-608d1365d776
> 
> Installed tedious instead of mysql2. Hope this helps.

I have done this and it is not that difficult If anyone wants to see a SOLID implementation  check out the "The road to graphql" by Robin Wieruch

Full walk though

**Comment by muloka** [comment]
Prisma 2 is currently being rewritten using Rust instead of Scala.

Are any more DB connectors being written for Prisma 1? Or is the plan to wait until Prisma 2 is out before any more are written? And at that point we may expect to have a SQL Server DB connector?

**Comment by AndryHTC** [comment]
Is it a good day to ask about ETA? Or even to ask about EST?
I've seen a lot of good feature proposal for Prisma 2! 😄

But... it's a shame that most of the people are cut off because of the lack of connectors

**Comment by toddtarsi** [comment]
Hey, for anyone trying to help out here, but hitting that 404:

It's not too bad to correct the link. Just change 'graphcool' to 'prisma' and the link will work. That will help, because the next link is full of links that 404 too 😆

**Comment by muloka** [comment]
> But... it's a shame that most of the people are cut off because of the lack of connectors

What problem are you trying to solve? Is it that you want to do GraphQL with SQL Server or is it that you want to specifically use Primsa with SQL Server?

**Comment by AndryHTC** [comment]
The final target

> @AndryHTC
> 
> > But... it's a shame that most of the people are cut off because of the lack of connectors
> 
> What problem are you trying to solve? Is it that you want to do GraphQL with SQL Server or is it that you want to specifically use Prisma with SQL Server?

The final target is obtaining a GraphQL with SQL Server, with subscriptions, schema migrations and all the good stuff of PRISMA. So, probably... The final target is obtaining PRISMA for SQL Server?

I've seen another great tool, also that only for Postgres [https://hasura.io/](Hasura)

**Comment by AndryHTC** [comment]
Sorry, but everything is a mess... Do we need to wait here the connector even for Prisma 2, Prisma Framework or Photon.js ecc...?

**Comment by thesobercoder** [comment]
This will be a game changer and I for one would be more than glad to encourage my team to adopting this.

**Comment by evanheckert** [comment]
Any insights into where this might fit on the roadmap?

**Comment by AndryHTC** [comment]
After 2+ years would be gold :)

**Comment by Sureiya** [comment]
Is there any way to post a bounty towards this feature?

**Comment by chuanqisun** [comment]
> Is there any way to post a bounty towards this feature?

Bounty or somehow Microsoft buys prisma 😂

**Comment by runar-indico** [comment]
For prisma 1, I do not think this will happen. For prisma 2, it is possible to start a project with this connector.

**Comment by DarkLite1** [comment]
Starting a new project and used `typeorm` for that. Until I saw a demo of Prisma and fell in love with its simple and clean syntax. God I would love to use it in cooperation with MSSQL. Waiting eagerly ... :)

**Comment by AndryHTC** [comment]
> Starting a new project and used `typeorm` for that. Until I saw a demo of Prisma and fell in love with its simple and clean syntax. God I would love to use it in cooperation with MSSQL. Waiting eagerly ... :)

Good luck :)

**Comment by muloka** [comment]
Prisma is working on this it seems. They've done a lot work patching this driver:

https://github.com/prisma/tiberius

If anyone knows Rust and wants to contribute their expertise on this then check out this repo.

**Comment by seyaobey** [comment]
Eagerly waiting on this. Because I need a sql database for my project, I'm currently using MySql and it sucks...
